### DISCUSSION TITLE
Add BBCode to Label3D

### DISCUSSION THREAD

**Discussion by elvisish** [godotengine/godot-proposals#8038] [3D]
### Describe the project you are working on

Walking simulator where I need to highlight certain letters in a Label3D.

### Describe the problem or limitation you are having in your project

Label3D has no way of changing the color on specific letters, that I know of.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

BBCode would be implemented as it is for RichTextLabel.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

It would just be possible to enable and use BBCode in Label3D.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Not that I know of.

### Is there a reason why this should be core and not an add-on in the asset library?

Not that I know of.

**Comment by Mickeon** [comment]
BBCode rendering is very complex and that's why there's a distinction between Label and RichTextLabel in the first place. For more advanced use-cases it may be worth to fetch the ViewportTexture from a Viewport that only contains the RichTextLabel inside.

**Comment by elvisish** [comment]
> BBCode rendering is very complex and that's why there's a distinction between Label and RichTextLabel in the first place. For more advanced use-cases it may be worth to fetch the ViewportTexture from a Viewport that only contains the RichTextLabel inside.

Thanks, I'll use that technique in the meantime, it would be convenient if it were possible though.

**Comment by Calinou** [comment]
You can also use the approach used in the Labels and Texts demo for laying out several Label3Ds relative to each other: https://github.com/godotengine/godot-demo-projects/blob/master/3d/labels_and_texts/label_3d_layout.gd

**Reply by elvisish** [reply]
I have many, many letters I need to re-color so that would be really difficult possibly with this approach, I wonder how performant using viewport textures is for projecting RichTextLabels in 3D space?
